Fails to compile on architectures where char is unsigned by default

Similar issue (maybe the same issue?) was fixed in the [Datastax/cpp-driver](https://github.com/datastax/cpp-driver) here: https://github.com/datastax/cpp-driver/pull/320 

```
#19 128.4 /opt/app/deps/erlcass/c_src/uuid_serialization.cc:44:1: error: narrowing conversion of '-1' from 'int' to 'char' [-Wnarrowing]
#19 128.4 /opt/app/deps/erlcass/c_src/uuid_serialization.cc: In function 'CassError erlcass::cass_uuid_from_string_n(const char*, size_t, CassUuid*)':
#19 128.4 /opt/app/deps/erlcass/c_src/uuid_serialization.cc:81:60: error: comparison is always false due to limited range of data type [-Werror=type-limits]
#19 128.4    81 |         if (kHexaToByteTable[static_cast<uint8_t>(pos[0])] == -1 || kHexaToByteTable[static_cast<uint8_t>(pos[1])] == -1)
#19 128.4       |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~
#19 128.4 /opt/app/deps/erlcass/c_src/uuid_serialization.cc:81:116: error: comparison is always false due to limited range of data type [-Werror=type-limits]
#19 128.4    81 |         if (kHexaToByteTable[static_cast<uint8_t>(pos[0])] == -1 || kHexaToByteTable[static_cast<uint8_t>(pos[1])] == -1)
#19 128.4       |                                                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~
#19 128.4  CPP    fun_bindings.cc
#19 128.5 cc1plus: all warnings being treated as errors
#19 128.5 make[1]: *** [nif.mk:84: /opt/app/deps/erlcass/c_src/uuid_serialization.o] Error 1
#19 128.5 make[1]: *** Waiting for unfinished jobs....
#19 130.3 make[1]: Leaving directory '/opt/app/deps/erlcass/c_src'
#19 130.3 make: *** [Makefile:44: nif_compile] Error 2
#19 130.3 ===> Hook for compile failed!
```

--- a/c_src/uuid_serialization.cc
+++ b/c_src/uuid_serialization.cc
@@ -23,7 +23,7 @@ const char* kByteToHexaTable[256] =
     "f0", "f1", "f2", "f3", "f4", "f5", "f6", "f7", "f8", "f9", "fa", "fb", "fc", "fd", "fe", "ff"
 };
 
-const char kHexaToByteTable[256] =
+const signed char kHexaToByteTable[256] =
 {
     -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,
     -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,

